From c9c3ff9f41bb02a24b1ff973e45ae58e6e44e39c Mon Sep 17 00:00:00 2001
From: Tin Tran <tin.tran.xk@renesas.com>
Date: Fri, 30 May 2025 03:57:26 +0700
Subject: [PATCH 39/46] Revert "PCI: dwc-plat: Add R-Car Gen5 PCIe 4.0 Host and
 Endpoint support"

This reverts commit 720656604ff2f9ea739bad0ab18693b7badc873e.

Removing the PCIe 4.0 on R-Car Gen5 from DW_Plat driver

Signed-off-by: Tin Tran <tin.tran.xk@renesas.com>
---
 .../pci/controller/dwc/pcie-designware-plat.c | 134 +-----------------
 1 file changed, 1 insertion(+), 133 deletions(-)

diff --git a/drivers/pci/controller/dwc/pcie-designware-plat.c b/drivers/pci/controller/dwc/pcie-designware-plat.c
index 805163f578c7..1fcfb840f238 100644
--- a/drivers/pci/controller/dwc/pcie-designware-plat.c
+++ b/drivers/pci/controller/dwc/pcie-designware-plat.c
@@ -20,107 +20,16 @@
 
 #include "pcie-designware.h"
 
-/* PCI Express capability */
-#define	EXPCAP(x)		(0x0070 + (x))
-
-#define	PCI_EXP_LNKCAP_MLW_X1	0x00000010 /* Maximum Link Width x1 */
-#define	PCI_EXP_LNKCAP_MLW_X2	0x00000020 /* Maximum Link Width x2 */
-#define	PCI_EXP_LNKCAP_MLW_X4	0x00000040 /* Maximum Link Width x4 */
-
-/* Renesas-specific */
-#define	PCIEMSR0		0x0000
-#define	BIFUR_MOD_SET_ON	BIT(0)
-#define	DEVICE_TYPE_EP		0
-#define	DEVICE_TYPE_RC		BIT(4)
-
-#define	PCIEINTSTS0EN		0x0310
-#define	MSI_CTRL_INT		BIT(26)
-
-#define to_rcar_gen5_pcie(x)	dev_get_drvdata((x)->dev)
-
 struct dw_plat_pcie {
 	struct dw_pcie			*pci;
 	enum dw_pcie_device_mode	mode;
-	void __iomem			*base;
-	void __iomem			*phy_base;
 };
 
 struct dw_plat_pcie_of_data {
 	enum dw_pcie_device_mode	mode;
 };
 
-void rcar_gen5_pcie_set_max_link_width(struct dw_plat_pcie *dw_plat_pcie, int num_lanes)
-{
-	struct dw_pcie *pci = dw_plat_pcie->pci;
-	u32 val;
-
-	val = dw_pcie_readl_dbi(pci, EXPCAP(PCI_EXP_LNKCAP));
-	val &= ~PCI_EXP_LNKCAP_MLW;
-	switch (num_lanes) {
-	case 1:
-		val |= PCI_EXP_LNKCAP_MLW_X1;
-		break;
-	case 2:
-		val |= PCI_EXP_LNKCAP_MLW_X2;
-		break;
-	case 4:
-		val |= PCI_EXP_LNKCAP_MLW_X4;
-		break;
-	default:
-		dev_info(pci->dev, "Invalid num-lanes %d\n", num_lanes);
-		break;
-	}
-	dw_pcie_writel_dbi(pci, EXPCAP(PCI_EXP_LNKCAP), val);
-}
-
-int rcar_gen5_pcie_set_device_type(struct dw_plat_pcie *dw_plat_pcie, bool rc,
-				   int num_lanes)
-{
-	u32 val;
-
-	/* Note: Assume the reset is asserted here */
-	val = readl(dw_plat_pcie->base + PCIEMSR0);
-	if (rc)
-		val |= DEVICE_TYPE_RC;
-	else
-		val |= DEVICE_TYPE_EP;
-	if (num_lanes < 4)
-		val |= BIFUR_MOD_SET_ON;
-	writel(val, dw_plat_pcie->base + PCIEMSR0);
-
-	return 0;
-}
-
-static int rcar_gen5_pcie_host_init(struct dw_pcie_rp *pp)
-{
-	struct dw_pcie *pci = to_dw_pcie_from_pp(pp);
-	struct dw_plat_pcie *dw_plat_pcie = to_rcar_gen5_pcie(pci);
-	int ret;
-	u32 val;
-
-	/* Set device type */
-	ret = rcar_gen5_pcie_set_device_type(dw_plat_pcie, true, pci->num_lanes);
-	if (ret < 0)
-		return ret;
-
-	dw_pcie_dbi_ro_wr_en(pci);
-
-	if (IS_ENABLED(CONFIG_PCI_MSI)) {
-		/* Enable MSI interrupt signal */
-		val = readl(dw_plat_pcie->base + PCIEINTSTS0EN);
-		val |= MSI_CTRL_INT;
-		writel(val, dw_plat_pcie->base + PCIEINTSTS0EN);
-	}
-
-	rcar_gen5_pcie_set_max_link_width(dw_plat_pcie, pci->num_lanes);
-
-	dw_pcie_dbi_ro_wr_dis(pci);
-
-	return 0;
-}
-
 static const struct dw_pcie_host_ops dw_plat_pcie_host_ops = {
-	.host_init = rcar_gen5_pcie_host_init,
 };
 
 static void dw_plat_pcie_ep_init(struct dw_pcie_ep *ep)
@@ -194,33 +103,6 @@ static int dw_plat_add_pcie_port(struct dw_plat_pcie *dw_plat_pcie,
 	return 0;
 }
 
-static int rcar_gen5_pcie_get_resources(struct dw_plat_pcie *dw_plat_pcie,
-					struct platform_device *pdev)
-{
-	struct resource *res;
-	struct dw_pcie *pci = dw_plat_pcie->pci;
-	struct device_node *np = dev_of_node(&pdev->dev);
-
-	of_property_read_u32(np, "num-lanes", &pci->num_lanes);
-
-	/* Renesas-specific registers */
-	dw_plat_pcie->base = devm_platform_ioremap_resource_byname(pdev, "app");
-	if (IS_ERR(dw_plat_pcie->base))
-		return PTR_ERR(dw_plat_pcie->base);
-
-	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "phy");
-	if (res) {
-		dw_plat_pcie->phy_base = devm_ioremap_resource(&pdev->dev, res);
-		if (IS_ERR(dw_plat_pcie->phy_base))
-			dw_plat_pcie->phy_base = NULL;
-	}
-
-	/* temporarily removed since PCIe on VDK Gen5 has not supported yet */
-	//return rcar_gen5_pcie_devm_reset_get(dw_plat_pcie, pci->dev);
-
-	return 0;
-}
-
 static int dw_plat_pcie_probe(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
@@ -249,12 +131,6 @@ static int dw_plat_pcie_probe(struct platform_device *pdev)
 	dw_plat_pcie->pci = pci;
 	dw_plat_pcie->mode = mode;
 
-	ret = rcar_gen5_pcie_get_resources(dw_plat_pcie, pdev);
-	if (ret < 0) {
-		dev_err(dev, "Failed to request resource: %d\n", ret);
-		return ret;
-	}
-
 	platform_set_drvdata(pdev, dw_plat_pcie);
 
 	switch (dw_plat_pcie->mode) {
@@ -297,20 +173,12 @@ static const struct of_device_id dw_plat_pcie_of_match[] = {
 		.compatible = "snps,dw-pcie-ep",
 		.data = &dw_plat_pcie_ep_of_data,
 	},
-	{
-		.compatible = "renesas,rcar-gen5-pcie",
-		.data = &dw_plat_pcie_rc_of_data,
-	},
-	{
-		.compatible = "renesas,rcar-gen5-pcie-ep",
-		.data = &dw_plat_pcie_ep_of_data,
-	},
 	{},
 };
 
 static struct platform_driver dw_plat_pcie_driver = {
 	.driver = {
-		.name	= "pcie4-rcar-gen5",
+		.name	= "dw-pcie",
 		.of_match_table = dw_plat_pcie_of_match,
 		.suppress_bind_attrs = true,
 	},
-- 
2.34.1

